### PR TITLE
Implement mount option support

### PR DESCRIPTION
**Pull request by laenion**
Implements #872, using the _mkfs_ option implementation as a reference.

With this new `mountOptions` entry it is possible to add mount options for each file system, used during the `mount` stage.

### CODE REVIEW

**Comment by ajeddeloh** [comment]
ok to test

File: internal/exec/stages/mount/mount.go [hunk 1]
```diff
@@ -117,10 +118,19 @@ func (s stage) mountFs(fs types.Filesystem) error {
...
+	args := translateOptionSliceToString(fs.MountOptions)
+	if err := s.Logger.LogOp(func() error { return syscall.Mount(fs.Device, path, *fs.Format, 0, args) },
+		"mounting %q at %q with type %q and options %q", fs.Device, path, *fs.Format, args,
 	); err != nil {
 		return err
 	}
 	return nil
 }
+
+func translateOptionSliceToString(opts []types.MountOption) string {
```
Comment (lucab) [inline internal/exec/stages/mount/mount.go]: Minor: it would be useful to take `,` as an argument here, and also have some basic tests to cover the empty-slice case and a simple populated-slice too.

**Comment by lucab** [comment]
PR and logic seems overall fine to me.
I have the fear that this will confuse a lot people, who will start assuming that they can put here options that will be in effect in the rootfs and for any subsequent boot.
How can we make obvious that these are just ephemeral options for Ignition and initramfs and first-boot only?

**Comment by laenion** [comment]
Thanks for the reviews and sorry for the late reply, I've been on vacation last week.

> I have the fear that this will confuse a lot people, who will start assuming that they can put here options that will be in effect in the rootfs and for any subsequent boot.
> How can we make obvious that these are just ephemeral options for Ignition and initramfs and first-boot only?

The most obvious solution would probably be to mention that to the documentation explicitly. Maybe by adding a sentence such as
> Please note that these options will only be applied during Ignition's `mount` stage (i.e. using this option will not result in automatic creation of corresponding mount units or fstab entries).

to the documentation of the `mountOptions` option?

I'll fix the remaining issues shortly.

File: internal/exec/stages/mount/mount.go (hunk 1 above)
Comment (laenion) [reply internal/exec/stages/mount/mount.go]: Will add the separator as a separate argument.

Regarding the slice: I'm no Go expert, but from what I know and see Go's `Join` function will handle both the empty and single slice automatically (i.e. return an empty string / a string without comma). Or do you mean testing the functionality via Blackbox tests?

**Comment by laenion** [comment]
Added the translator and added the option to the IgnoreDuplicates list. Also added the separator as a separate argument to `translateOptionSliceToString`.

File: internal/exec/stages/mount/mount.go [hunk 2]
```diff
@@ -117,10 +118,19 @@ func (s stage) mountFs(fs types.Filesystem) error {
 		return err
 	}
 
-	if err := s.Logger.LogOp(func() error { return syscall.Mount(fs.Device, path, *fs.Format, 0, "") },
-		"mounting %q at %q with type %q", fs.Device, path, *fs.Format,
+	args := translateOptionSliceToString(fs.MountOptions, ",")
+	if err := s.Logger.LogOp(func() error { return syscall.Mount(fs.Device, path, *fs.Format, 0, args) },
```
Comment (ajeddeloh) [inline internal/exec/stages/mount/mount.go]: Sorry for the late review; I was at a conference then some regressions prevented me from building an image to test for a bit. Looks like this only works with FS specific options, but not ones that end up as flags for the fourth argument. For example, `ikeep` is xfs specific and works, but `noexec` is not and doesn't since that ends up as the flag `MS_NOEXEC`. So this needs to learn how to handle those separately. It would be good to add a blackbox test for both types of args as well.

Comment (ajeddeloh) [reply internal/exec/stages/mount/mount.go]: https://github.com/karelzak/util-linux/blob/master/libmount/src/optstr.c#L651 and that file in general looks useful, though it would need cgo.

Comment (laenion) [reply internal/exec/stages/mount/mount.go]: Just that it's not part of Go, but an additional dependency. I wasn't aware that `syscall` is deprecated, so this is indeed a non-issue.

Comment (ajeddeloh) [reply internal/exec/stages/mount/mount.go]: It'd be good to not vendor something not really meant as a library especially something large like docker, so I think I'd prefer exec'ing mount if you're opposed to the cgo implementation. @lucab do you have strong opinions here?

Comment (laenion) [reply internal/exec/stages/mount/mount.go]: I'm not opposed to use _libmount_, it's just that I usually try to find a native library before implementing something myself.

FYI: The size of the Ignition executable is increasing by ~100 KB when using the Docker module.

Comment (ajeddeloh) [reply internal/exec/stages/mount/mount.go]: It's less about the actual binary size and more about needing to keep up to date with another dependency as well as the go module tooling being bad and wanting to pull in more than necessary to `vendor/`.

Comment (lucab) [reply internal/exec/stages/mount/mount.go]: @ajeddeloh either directions are fine. As we are already shelling out to `mount` in other places, I think that would be path of least resistance here too.

Comment (laenion) [reply internal/exec/stages/mount/mount.go]: Implemented as a shell command as suggested.

**Comment by laenion** [comment]
Added a positive and a negative Blackbox test case for the mount stage.

File: tests/negative/filesystems/mount_filesystem.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
+// Copyright 2017 CoreOS, Inc.
```
Comment (ajeddeloh) [inline tests/negative/filesystems/mount_filesystem.go]: 2019 Red Hat

File: tests/positive/filesystems/mount_filesystem.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
+// Copyright 2017 CoreOS, Inc.
```
Comment (ajeddeloh) [inline tests/positive/filesystems/mount_filesystem.go]: 2019 Red Hat
